### PR TITLE
examples/llama+neuron: enable buffer donation

### PR DESCRIPTION
**Pull request by Corendos**
Donate verything except the `token_index` buffer as it messes up the output

### CODE REVIEW

File: examples/llama/llama.zig [hunk 1]
```diff
@@ -370,12 +370,12 @@ pub const KvCache = struct {
                 .{ .layer = self.layer_index, .k = token_index },
                 // transpose to match kv-cache layout
                 new_k.contiguous(.{ .h, .k, .hd }),
-            ).reuseBuffer(self.k),
```
[the remark this answers is not in the document]
Comment (Corendos) [reply examples/llama/llama.zig:373]: We still have `reuseBuffer` on `KvCache` and we use this instead

[the remark this answers is not in the document]
Comment (gwenzek) [reply examples/llama/llama.zig:373]: then can you keep this one ? it's safer to donate twice than to not donate at all.

**Review by gwenzek — COMMENTED** [review]
approving, but I have some questions:

can you try to reduce the source of the problem ? 

if you compile just the `increment` kernel with donations what happens ?
is the problem due to the dtype ? the shape ?

**Review by gwenzek — APPROVED** [review]

File: examples/llama/llama.zig (hunk 1 above)
[the remark this answers is not in the document]
Comment (Corendos) [reply examples/llama/llama.zig]: Addressed in https://github.com/zml/zml/pull/112/commits/0ba4ee898e39951706d5a28d356de7fc3555a1ed

**Pull request merged**